src/test/ui/wf/wf-static-method.stderr

```diff
@@ -20,8 +20,9 @@ LL | impl<'a, 'b> Foo<'a, 'b, ()> for IndirectEvil<'a, 'b> {
    |      lifetime `'a` defined here
 ...
 LL |         let me = Self::make_me();
-   |                  ^^^^^^^^^^^^^ requires that `'b` must outlive `'a`
+   |                  ^^^^^^^^^^^^^ type annotation requires that `'b` must outlive `'a`
```

Comment:
There is really no type annotation here :)

Reply:
My reasoning is that `Self` is considered a type annotation, and the subsequent help note should make it clear.

Reply:
I can change it to `use of Self requires ...` but this may produce false positives/negatives in some edge cases because the heuristics this relies on is not 100% reliable. This might not be ok given that this is the primary label of the error, but I'm not sure.

Reply:
I think calling Self a type annotation is a bit of a stretch. I'd like to see what the diagnostics look like with the change.